cmd/podman/pods/create.go

```diff
@@ -104,6 +103,18 @@ func create(cmd *cobra.Command, args []string) error {
 		createOptions.Share = nil
 	} else {
 		createOptions.Share = strings.Split(share, ",")
+		if cmd.Flag("infra-command").Changed {
+			createOptions.InfraCommand, err = cmd.Flags().GetString("infra-command")
```

Comment:
Why is that necessary? It seems rather exotic. A comment would be very helpful.

Reply:
Same below.

Reply:
Not sure it is exotic, since this type of code is all over the cli.  Basically we want to make sure the user made a change using the CLI, rather then sending the default settings over to the server.  That way the server can differentiate if the user wanted a change or just to take defaults.

Reply:
So we wouldn't take the defaults from the client containers.conf but from the server?

Reply:
Yes

Reply:
What's the reasoning for that? Do we document that somewhere?

I think there's some inconsistency among the commands w.r.t. to which defaults come from the client and server.

I am not against the idea but just want to understand if we have some policy for that decision.

Reply:
All defaults should come from the server.  Any defaults that come from the client are a bug.  But I agree there are some coming from the client.

Reply:
@vrothberg I now have comments, PTAL